### DISCUSSION TITLE
access gpiomem with user

### DISCUSSION THREAD

**Discussion by patcmorneau** [WiringPi/WiringPi#285] [Q&A] [answered]
Hello,
 I am doing development on a raspberry pi 4 running ubuntu 20.04.
 I installed wiringpi via the apt. The package seems to be at least 3 years old.
 
I have an issue where if i run my program with the current user and provided the user the access to /dev/gpiomem, It crashes my cable network interface.

[this](https://raspberrypi.stackexchange.com/a/46235) might explain why this is happening.

So I am wondering if i compile a newer version of wiringpi if this will fix the issue.

**Comment by mstroh76** [comment, marked as answer]
Ubuntu is currently not actively supported. However, this is in progress or planned. The current version was last tested on Ubuntu 24.04 LTS (Raspberry Pi 5) and worked. Version 3.8 is preferred here. Older Ubuntu versions than 24.04 are not tested.

Additionally, the current version can also access the GPIOs via the user space with Character Device Userspace API (gpiod) , ths is done with the function wiringPiSetupGpioDevice .
